File: datasets/lua-keep-incoder_6b_temp_0.2-0/HumanEval_27_flip_case.lua
-- For a given string, flip lowercase characters to uppercase and uppercase to lowercase.
-- >>> flip_case('Hello')
-- 'hELLO'
local function flip_case(string)
  local result = string
  result:gsub('%l', function(c)
    return string.char(tonumber(c) + 32)
  end)
  result:gsub('%U', function(c)
    return string.char(tonumber(c) - 32)
  end)
  return result
end


lu = require('luaunit')

function test_humaneval()
local candidate = flip_case
    lu.assertEquals(candidate(''), '')
    lu.assertEquals(candidate('Hello!'), 'hELLO!')
    lu.assertEquals(candidate('These violent delights have violent ends'), 'tHESE VIOLENT DELIGHTS HAVE VIOLENT ENDS')
end

os.exit(lu.LuaUnit.run())

File: datasets/cpp-keep-code_davinci_001_temp_0.2/HumanEval_27_flip_case.cpp
#include<assert.h>
#include<bits/stdc++.h>
// For a given string, flip lowercase characters to uppercase and uppercase to lowercase.
// >>> flip_case('Hello')
// 'hELLO'
std::string flip_case(std::string string) {
    std::string result;
    for (int i = 0; i < string.length(); i++) {
        if (string[i] >= 'A' && string[i] <= 'Z') {
            result += string[i] + 32;
        } else if (string[i] >= 'a' && string[i] <= 'z') {
            result += string[i] - 32;
        } else {
            result += string[i];
        }
    }
    return result;

}
int main() {
    auto candidate = flip_case;
    assert(candidate(("")) == (""));
    assert(candidate(("Hello!")) == ("hELLO!"));
    assert(candidate(("These violent delights have violent ends")) == ("tHESE VIOLENT DELIGHTS HAVE VIOLENT ENDS"));
}
